### DISCUSSION TITLE
[Question] Line color depending on the entry

### DISCUSSION THREAD

**Discussion by raudonikis** [patrykandpatrick/vico#576] [Ideas]
(Jetpack Compose)
Hi, great library! One thing I could not figure out yet - is there any way to set the line color (line chart) depending on some condition on the entry model? My use case is that I have a data set of dates (days of the month if more precisely), and I want to show a different color for future days, because they do not have any data yet.

**Comment by patrickmichalik** [comment]
Hello! Unfortunately, there’s no such feature at the moment. You could try adding two data sets to your `ChartEntryModel`, one for either line color, and using two `LineSpec`s. However, it’s possible that difference animations wouldn’t work as you’d like them to. Another option is to create a custom line chart implementation, copying over the `LineChart` class and making the desired changes. We’ll soon make classes such as this one open for extension.

**Comment by raudonikis** [comment]
I see, that's a shame, I'll try adding multiple data sets, that might work. Thank you for a great library and quick responses to all my recent questions 😄

**Comment by patrickmichalik** [comment]
Thank you for the kind words, and you’re welcome! As it turns out, `LineChart` is already open for extension, so you should be able to extend it and make the necessary overrides. However, there are several functions whose visibility is `private` instead of `protected`, so you’d potentially have to copy those over. We’ll update their visibility to make extension simpler.

**Comment by patrickmichalik** [comment]
Hello! Vico 1.6.5 introduces the aforementioned visibility changes.

**Comment by sarthak-greenely** [comment]
Hey @patrickmichalik, I was looking into how this can be achieved, and looks like its still not very straightforward to do it, I can extend the LineChart class, but still need to copy/paste most of the drawing logic for example. 🤔 (let me know if there is a simpler way to do it though)
I was thinking if the library could expose a function which would draw the lineChart point by point on canvas (something similar to PointConnector). rather than drawing the whole Path at once, do you think something like that could be a good idea?

p.s: Thanks for the amazing library.

**Comment by patrickmichalik** [comment]
Thanks for the kind words, @sarthak-greenely! You’re right—you might have to copy over quite some code. We plan on making it possible to style lines on a segment-by-segment basis. We’ve not designed the API yet, but it’ll most likely be a `LineSpec`-mutating lambda. This is quite similar to what you’ve described, only a bit higher-level (and thus slightly simpler to use).
